klog.go

```diff
@@ -699,7 +699,11 @@ func (buf *buffer) someDigits(i, d int) int {
 }
 
 func (l *loggingT) println(s severity, logger *logr.Logger, filter LogFilter, args ...interface{}) {
-	buf, file, line := l.header(s, 0)
+	l.printlnDepth(s, logger, filter, 1, args...)
```

Comment:
I'm seeing off-by-one errors  during stack unwinding after updating to a klog with this change.

Why is this 1 and not something else? Was that tested?

Reply:
I can change that to 2 without any test failing. Looks like we don't have any test for this in klog itself.

Reply:
sorry, I assumed that 1 for stack unwinding is correct value due to  

`
func Exitf(format string, args ...interface{}) {
  atomic.StoreUint32(&fatalNoStacks, 1)
  logging.printf(fatalLog, logging.logr, logging.filter, format, args...)
}
`

Reply:
and also by this: https://github.com/s3rj1k/klog/blame/868bc68c8b1236129dd305d801e01e076a033e39/klog.go#L721

Reply:
It breaks in Kubernetes when testing the JSON logger, i.e. when SetLogger is used with a Logger implementation that itself does stack unwinding. We can't test that in klog with just a unit test because we must not depend on such a logger.